File: Iniciante/1018 - Cédulas/TypeScript/1018.ts
import * as fs from "fs";
const BANKNOTES: number[] = [100, 50, 20, 10, 5, 2, 1];

function main(): void {
  const input: string = fs.readFileSync("/dev/stdin", "utf8");
  const value: number = parseInt(input);
  console.log(value);
  calculateBanknotes(value);
}

function calculateBanknotes(value: number): void {
  for (let banknote of BANKNOTES) {
    const amountBanknotes: number = Math.trunc(value / banknote);
    printQuantityBanknotes(amountBanknotes, banknote);
    value %= banknote;
  }
}

function printQuantityBanknotes(amountBanknotes: number, banknoteValue: number): void {
  console.log(`${amountBanknotes} nota(s) de R$ ${banknoteValue},00`);
}

main();

File: Iniciante/1018 - Cédulas/Python/1018.py
BANKNOTES = [100, 50, 20, 10, 5, 2, 1]


def main():
    value = int(input())
    print(value)
    calculate_banknotes(value)


def calculate_banknotes(value):
    for banknote in BANKNOTES:
        amount_banknotes = value // banknote
        print_quantity_banknotes(amount_banknotes, banknote)
        value %= banknote


def print_quantity_banknotes(amount_banknotes, banknoteValue):
    print(f"{amount_banknotes} nota(s) de R$ {banknoteValue},00")


if __name__ == "__main__":
    main()
